### DISCUSSION TITLE
Unable to connect to some machines outside of LAN

### DISCUSSION THREAD

**Discussion by Affection4036** [rustdesk/rustdesk-server-pro#325] [Q&A] [answered]
Hello,

I bought Rustdesk a about 2-3 weeks ago to start supporting new customers. I want to set-up unattended access but I am unable to connect to the machines on their netwerk. We do have full access to all network equipment and are able to make changes if necessary. 
We've checked our network equipment and it is set-up to pass all traffic on all ports. We also asked the ISP of the customer if they are blocking any ports or traffic. They checked their ACL but nothing is blocked.

We have a Rustdesk Pro licentie and installed the software on a separate machine hosted in a datacenter. It's hosted on a Ubuntu Server VM. The installation was done through the easy installation script shared on the Rustdesk website. For the API we use a HTTPS connection and use a domain name to connect to it.
When a colleague installed the Rustdesk client on their PC we were enable to connect. The machine is NOT on our LAN and the connection is logged on our portal.

When we attempt to connect to the machines I get the following message on my Mac, running Sanoma:

![image](https://github.com/user-attachments/assets/d9e01208-d6ca-4a72-9872-6170f4e9ec28)
Deadline has elapsed.

I think this message states that there is a timeout in the connection, but there is no log entry created on the web portal of our Rustdesk instance. I can't seem to find a local log file to check what's happening, so the only thing I have is the error message.

How can we troubleshoot/solve this?

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/wiki/FAQ#deploy-rustdesk-server-in-intranet

**Reply by Affection4036** [reply]
> https://github.com/rustdesk/rustdesk/wiki/FAQ#deploy-rustdesk-server-in-intranet

Just want to make sure: the machine is hosted in a datacenter that is in no way connected to this customer. My colleague is connected to a different network (not related with the datacenter or customer network) and Rustdesk works. Are you sure it's a NAT Hairpin problem if it's only this customers' network? The Rustdesk server is NOT hosted on the customers' site!

**Reply by rustdesk** [reply, marked as answer]
> datacenter

Yes, you are right, your problem has nothing to do with NAT Hairpin. It seems more related to https://github.com/rustdesk/rustdesk/wiki/FAQ#connection-reset-by-the-peer

> but there is no log entry created on the web portal of our Rustdesk instance.
This means the connection is not established yet.

Three logs we can check.

- The output of `hbbr`
- The log of controlled side, https://github.com/rustdesk/rustdesk/wiki/FAQ#access-logs
- The log of controlling side, on mac it is `~/Library/Logs/RustDesk/RustDesk_rCURRENT.log`

**Reply by Affection4036** [reply]
> > datacenter
> 
> Yes, you are right, your problem has nothing to do with NAT Hairpin. It seems more related to https://github.com/rustdesk/rustdesk/wiki/FAQ#connection-reset-by-the-peer

Forcing it to use the relay server by adding '/r' behind the ID seems to solve the issue. When using a Mac, is there a way to force every connection through the relay? When adding or editing the PC as a contact there is no way to force it.

**Reply by rustdesk** [reply]
![image](https://github.com/user-attachments/assets/caca304c-acaa-4aeb-aa45-c1055b956ff3)

**Reply by Affection4036** [reply]
I see, the option is found within the web console. I've enabled it and it seems to work. Thanks for the support!
